src/neurostatslib/solver.py

```diff
@@ -0,0 +1,593 @@
+"""
+## A Module for Optimization with Various Regularizations.
+
+This module provides a series of classes that facilitate the optimization of models
+with different types of regularizations. Each solver class in this module interfaces
+with various optimization methods, and they can be applied depending on the model's requirements.
+"""
+import abc
+import inspect
+from typing import Callable, List, Optional, Tuple, Union
+
+import jax.numpy as jnp
+import jaxopt
+from numpy.typing import NDArray
+
+from .base_class import _Base
+from .proximal_operator import prox_group_lasso
+
+__all__ = ["UnRegularizedSolver", "RidgeSolver", "LassoSolver", "GroupLassoSolver"]
+
+
+def __dir__() -> list[str]:
+    return __all__
```

Comment:
as an aside, probably worth describing this `__all__` / `__dir__` trick in the template repo